.github/workflows/run-tests.yml

```diff
@@ -173,6 +225,13 @@ jobs:
         timeout-minutes: 15
         run: UNIT_TEST_DIRS=$UNIT_TEST_DIR make unit-test-coverage
 
+      - name: Flaky Unit Test Detection
```

Comment:
I think we can collapse this and the two steps above into a single step with different arguments.